EnqueueMigrateServicesPreferencesActivity/handler.ts

```diff
@@ -0,0 +1,54 @@
+import { Context } from "@azure/functions";
+import { QueueServiceClient } from "@azure/storage-queue";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as e from "fp-ts/lib/Either";
+import * as te from "fp-ts/lib/TaskEither";
+import { MigrateServicesPreferencesQueueMessage } from "../MigrateServicePreferenceFromLegacy/handler";
+
+const LOG_PREFIX = "EnqueueMigrateServicesPreferencesActivity";
+
+type IEnqueueMigrateServicesPreferencesActivityHandler = (
+  queueService: QueueServiceClient,
+  queueName: NonEmptyString
+) => (context: Context, rawInput: unknown) => Promise<string>;
+
+export const getEnqueueMigrateServicesPreferencesActivityHandler: IEnqueueMigrateServicesPreferencesActivityHandler = (
+  queueService,
+  queueName
+) => async (context: Context, rawInput: unknown): Promise<string> =>
+  te
+    .fromEither(MigrateServicesPreferencesQueueMessage.decode(rawInput))
+    .foldTaskEither(
+      error => {
+        context.log.error(
+          `${LOG_PREFIX}|Cannot parse input|ERROR=${readableReport(error)}`
+        );
+        return te.fromEither(e.right("FAILURE"));
+      },
```

Comment:
why folding TE here instead of mapping left path in the previous `decode` ?

Reply:
becouse the fold let me remove the decode error from the taskEither left side. In this way the decode error will be consdiered as a "FAILURE" and not an Error, without trigger useless retry.
With the mapLeft it will still be left sided.